packages/button/src/Button.ts

```diff
@@ -51,6 +54,42 @@ export class Button extends SizedMixin(ButtonBase, { noDefaultSize: true }) {
         return [...super.styles, buttonStyles];
     }
 
+    protected pendingCooldown = -1;
+
+    constructor() {
+        super();
+        this.addEventListener(
+            'click',
+            (event: Event): void | boolean => {
+                if (this.pending) {
+                    event.preventDefault();
+                    event.stopImmediatePropagation();
+                    event.stopPropagation();
+                    return false;
+                }
+            },
+            {
+                capture: true,
+            }
+        );
+    }
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-pending' })
+    public isPending = false;
+
+    @property({ type: String, attribute: 'pending-label' })
+    public pendingLabel = 'Pending';
+
+    @property({ type: Boolean, reflect: true, attribute: true })
+    public pending = false;
```

Comment:
I don't really understand the use of `pending` and `isPending`. Can you add some comments to clarify the workflow that's being acheived?

Reply:
Yeah `pending` is the indication that this is a pending state button, while `is-pending` is added/removed during the pending state. Agree needs comments.

Reply:
That timing seems weird, though it does make the code make so much more sense.

Based on this [documentation](https://spectrum.adobe.com/page/button/#Pending), it feels like `pending` should be an immediate state. Was there something specific to your application requirements that lead to this relationship?

Reply:
`pending` is the property that the user supplies to the component upon which we update the `isPending` state variable to true and do the necessary css updates. `isPending` represents the state of the component and can be accessed through DOM Apis.

Reply:
Well it was more a matter of convenience at point of use. If the `pending` state is added as an indicator of future behavior, then the click handler behaves like a normal button, with the click event firing on first click. The button then adds its own `isPending` state to track its timeout period and disable further clicks during that time.

If the consumer has to add the `pending` state when click occurs, thats a little more work for the consumer to deal with? Not sure, I could go both ways on this, but it felt convenient to have it this way, and it marks the button in the consumers render template as a pending state button so its quite clear to see that the behavior will be distinct here, rather than a side effect of the buttons click handler?

Reply:
This is an interesting approach, but not seemingly what the code says?
```
        if (changedProperties.has('pending')) {
            if (this.pending) {
                this.pendingCooldown = window.setTimeout(() => {
                    if (!this.disabled) {
                        this.isPending = true;
                        this.setAttribute('aria-label', this.pendingLabel);
                    }
                }, 1000);
            } else if (this.pending === false && this.isPending) {
                window.clearTimeout(this.pendingCooldown);
                this.isPending = false;
                this.setAttribute('aria-label', this.label || '');
            }
        }
```
This may be updated functionality, but this says that as soon as you set `pending` you start the clock for `isPending`. 

I discussed why this might be in with the design team, and they outlined a _seemingly_ valid use case of no wanting to go to "visually pending" and then "processed" in flicker speed progression. It's valid, but it's valid at any point in the process where you might switch to `pending`, which makes me wonder if it. would be better to require an application manage any staged loading they do local to where there can be some understanding of the time required to do the actual task in question.

I can be convinced of either way, with clear comments that outline a workflow that we all see the same way, but right now I'm leaning towards it being better to only offer `pending` and it being immediate.

Reply:
Ah yes sorry forgot that nuance (its been a while since i started this one)....

I think if we leave management of the visually pending state to the consumer, like as not it will get implemented poorly in many places and result in a multitude of flickering bugs in a variety of places with solutions to it varying across the board.

But yes the state flow as I recall now is:

* Enabled button
* Click
* Click handler - consumer assigns `pending` state
* `isPending` timeout begins
* `isPending` && `pending` causes display of 'visually pending' state
* Time goes by
* Consumer removes `pending` state
* `isPending` is removed
* Enabled button.

Reply:
Yeah, this is ugly pretty much regardless... @TarunAdobe can you test this approach which would simplify to a single state variable.

- Enable button
- Click
- Consumer assignes `pending`
- The 'visually pending' state is applied via an immediate transition or animation that has a delay of `1000ms` but is customizable via a CSS Custom Property
- Time goes by
- Consumer removes `pending`
- The 'visually pending' state is removed via an immediate transition or animation that has no delay
- Button is enable

Reply:
done